console/console-init/ui/src/modules/address-space/components/EndpointConfiguraion/EndpointConfiguraion.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React from "react";
+import {
+  Grid,
+  GridItem,
+  Form,
+  Checkbox,
+  FormGroup,
+  Radio,
+  Switch
+} from "@patternfly/react-core";
+import { IMessagingProject } from "modules/address-space/dialogs";
+import {
+  tlsCertificateOptions,
+  endpointProtocolOptions,
+  TlsCertificateType
+} from "modules/address-space/utils";
+
+interface IEndpointConfigurationProps {
+  projectDetail: IMessagingProject;
+  setProjectDetail: (project: IMessagingProject) => void;
+}
+const EndpointConfiguration: React.FunctionComponent<IEndpointConfigurationProps> = ({
+  projectDetail,
+  setProjectDetail
+}) => {
+  const { tlsCertificate, protocols } = projectDetail;
+  const onRouteChange = (value: boolean) => {
+    if (!value) {
+      setProjectDetail({
+        ...projectDetail,
+        tlsTermination: undefined,
+        hostname: undefined,
+        addRoutes: value
+      });
+    } else {
+      setProjectDetail({ ...projectDetail, addRoutes: value });
+    }
+  };
+  const onCertificateChange = (_: boolean, event: any) => {
+    const value = event.target.value;
+    if (value) {
+      if (value === TlsCertificateType.UPLOAD_CERT) {
+        setProjectDetail({
+          ...projectDetail,
+          tlsCertificate: value,
+          addCertificate: true
+        });
+      } else {
+        setProjectDetail({
+          ...projectDetail,
+          tlsCertificate: value,
+          addCertificate: false
+        });
+      }
+    }
+  };
+  const findPrototcol = (protocol: string) => {
+    if (protocols) {
+      const filteredProtocols = protocols.filter(pr => pr === protocol);
+      return filteredProtocols[0];
+    }
+  };
+  const isProtocolPresent = (protocol: string) => {
+    if (protocols) {
+      if (findPrototcol(protocol)) {
+        return true;
+      }
+    }
+    return false;
+  };
+  const onProtocolChange = (checked: boolean, event: any) => {
+    const protocolValue = event.target.value;
+    let protocolList: string[] = [];
+    if (protocols) {
+      protocolList = JSON.parse(JSON.stringify(protocols));
+    }
+    if (protocolValue) {
+      if (checked) {
+        protocolList.push(protocolValue);
+      } else {
+        const index = protocolList.findIndex(pr => pr === protocolValue);
+        if (index >= 0) {
+          protocolList.splice(index, 1);
+        }
+      }
+    }
+    setProjectDetail({ ...projectDetail, protocols: protocolList });
+  };
+  return (
+    <Grid>
+      <GridItem span={6}>
+        <Form>
+          <FormGroup
+            fieldId="form-group-endpoint-protocol"
+            label="Protocol"
+            isRequired={true}
+          >
+            {endpointProtocolOptions.map(protocol => (
+              <>
```

Comment:
This will throw a warning because key is missing, use React Fragments in this case and provide a key.